Downloader.swift

```diff
@@ -0,0 +1,100 @@
+//
+//  downloadModel.swift
+//  react-native-spokestack
+//
+//  Created by Timothy Willison on 12/7/20.
+//
+
+import Foundation
+import Network
+
+@objc
+public class Downloader: NSObject {
+    private var allowCellular: Bool = false
+    private var refreshModels: Bool = false
+    private let monitor = NWPathMonitor()
+    private var path: NWPath?
+    private var downloadQueue: [URL:(Error?, String?) -> Void] = [:]
+    lazy var pathUpdateHandler = { (path: NWPath) in
+        self.path = path
+        if self.isNetworkAvailable() && !self.downloadQueue.isEmpty {
+            print("Downloader: executing from pathUpdateHandler")
+            for (url, complete) in self.downloadQueue {
+                self.downloadModel(url, complete)
+            }
+            // Reset queue
+            self.downloadQueue = [:]
```

Comment:
Side-effecting properties should be avoided, so this might be better broken up into a func and a property. Another option, since this is just set on init, is to simply move this entire block as the value of `monitor.pathUpdateHandler` in `init`

Reply:
I don't follow. The queue is reset when all downloads are triggered.

Reply:
I went ahead and moved the downloading to another function, but it still needs to reset the download queue, not matter where the code is.